Core/HLE/sceKernelInterrupt.cpp

```diff
@@ -826,6 +826,63 @@ static u32 sysclib_strncpy(u32 dest, u32 src, u32 size) {
 	return hleLogSuccessX(SCEKERNEL, dest);
 }
 
+static u32 sysclib_strtol(u32 t, u32 endStringtemp, int base) {
+	ERROR_LOG(SCEKERNEL, "Unimp sysclib_strtol(%08x, %08x, %i)", t ,endStringtemp, base);
+	// base == 0 seems to be handled as base == 10.
+	if (base == 0){
+		base = 10;
+	}
+	std::string str = Memory::GetCharPointer(t);
+	std::string endString = Memory::GetCharPointer(endStringtemp);
```

Comment:
if `endStringtemp` is a pointer to `char*` (ie. `char**`) this will probably pointing to the wrong address, right?

Reply:
Normally, `strtol()` writes to endPtr (if is is specified other than 0) with the pointer of the end of the string.

We could calculate this given the native strtol and its different from the input, i.e.:
```c++
char *str = Memory::GetCharPointer(strPtr);
char *end = nullptr;
int result = strtol(str, &end, base);
if (Memory::IsValidRange(endPtrPtr, 4))
    Memory::WriteUnchecked_U32(strPtr + (end - str), endPtrPtr);
```

That is, we can just get the distance from str (input arg) to end (output from `strtol()`) and add that to the PSP pointer, since it'd be the same distance.  Then we can write that as the PSP end pointer to where the PSP code asked for it.

-[Unknown]

Reply:
MSVC 2019 get  error
A value of type C++ cannot be used to initialize an entity of type
char* str = Memory::GetCharPointer(strPtr);
I tried const char str_[] = Memory::GetCharPointer(strPtr); still get error

Reply:
solved by
char *str
str = Memory::GetCharPointer(strPtr);